captcha-plugin-draft.txt

```diff
@@ -107,6 +107,13 @@ We provisionally instantiate these with
 
 [TODO: I would really like to use something other than RSA, but haven't found
 satisfactory research on alternative blind signature schemes.]
+[Alex: It could be possible to use a DSA-style blind signature (see
+http://blog.cryptographyengineering.com/p/note-on-blind-signature-schemes.html).
+The protocol requires three steps, the additional step is required since the
+server initially sends a random element for each signature that needs to be
+blindly signed. While this is not ideal it may be possible to incorporate
+this into the structure below so that the plugin asks for 10000 elements of
+randomness from the server when it generates the nonces.]
```

Comment:
I think Tanja Lange's recommendation to use RSA is becoming clearer now : 

I asked my local contact for pairing related questions.  Appears one should always expect similar performance characteristics from pairing base schemes and RSA, so that's largely irrelevant.  RSA gives the mint flexibility in keysize though. 

An article _On the Security of One-Witness Blind Signature Schemes_ by Foteini Baldimtsi and Anna Lysyanskaya seemingly proves that _"current techniques for proving security in the random oracle model do not work for the Schnorr blind signature"_.  Just peaking at a few paragraphs suggests that the extra-round is linked to this rather theoretical shortcoming.